### DISCUSSION TITLE
Using EmguCV 4.6.0 or later on Raspberry Pi

### DISCUSSION THREAD

**Discussion by TonyCongqianWang** [emgucv/emgucv#946] [Q&A] [answered]
I added the latest EmguCV nuget Package and runtime in my projekt and it runs on both windows and raspberry OS. But only yesterday I saw the installation guide and realized that I messed up. In windows, having the runtime nuget is enough, but on raspberry OS you are supposed to build emgu from git... I did not. I installed the OpenCV package from apt-get and somehow so far it worked out and I could build and run the same C# projekt on my Pi.

So my question is: was I just lucky? If I am not mistaken, the latest release with Raspberry Pi as supported platform is 4.5.5. Do I have to downgrade my nuget package?

**Comment by canmingh** [comment]
For the latest Raspberry Pi OS, you can use this runtime nuget package (instead of Emgu.CV.runtime.windows):

https://www.nuget.org/packages/Emgu.CV.runtime.debian-arm64#readme-body-tab

**Reply by TonyCongqianWang** [reply]
I don't quite understand. The nuget package only contains the libextern.so file but not the entire library if I am not mistaken.I still need to build the correct opencv library with the repository, right?

**Reply by emgucv** [reply]
All the other dlls will be installed by buget dependency resolution. If you install the Debian arm64 runtime package, the Emgu.CV nuget package will be included by dependency.

**Reply by TonyCongqianWang** [reply]
So if I just install the nuget packages I do not need to follow https://www.emgu.com/wiki/index.php/Download_And_Installation to run emguCV?

**Comment by TonyCongqianWang** [comment]
I have compiled the cvextern library myself and noticed that all dependencies were installed by apt-get. Does that mean that the cvextern lib given by the runtime independend of any cmake build after all? Download and Installation Guide did suggest that the only way to use emgu on linux was to clone the repo and compile everything. Is that not true after all and downloading the pre compiled runtime nuget is sufficient?

**Reply by TonyCongqianWang** [reply]
```linux-vdso.so.1 (0x00007ffd519d7000)
        libgeotiff.so.5 => /lib/x86_64-linux-gnu/libgeotiff.so.5 (0x00007f4a3ba9f000)
        libusb-1.0.so.0 => /lib/x86_64-linux-gnu/libusb-1.0.so.0 (0x00007f4a3ba83000)
        libopenjp2.so.7 => /home/mabrivision/anaconda3/lib/libopenjp2.so.7 (0x00007f4a3ba1f000)
        libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007f4a3b8e2000)
        libXt.so.6 => /lib/x86_64-linux-gnu/libXt.so.6 (0x00007f4a3b679000)
        libOpenGL.so.0 => /lib/x86_64-linux-gnu/libOpenGL.so.0 (0x00007f4a3b64d000)
        libGLX.so.0 => /lib/x86_64-linux-gnu/libGLX.so.0 (0x00007f4a3b617000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f4a3b5f4000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f4a3b5ee000)
        libstdc++.so.6 => /home/mabrivision/anaconda3/lib/libstdc++.so.6 (0x00007f4a3b479000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f4a3b32a000)
        libgcc_s.so.1 => /home/mabrivision/anaconda3/lib/libgcc_s.so.1 (0x00007f4a3b316000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4a3b122000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f4a42033000)
        libproj.so.15 => /lib/x86_64-linux-gnu/libproj.so.15 (0x00007f4a3ae33000)
        libtiff.so.5 => /lib/x86_64-linux-gnu/libtiff.so.5 (0x00007f4a3adb1000)
        libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007f4a3ad84000)
        libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f4a3ad5a000)
        libSM.so.6 => /lib/x86_64-linux-gnu/libSM.so.6 (0x00007f4a3ad4f000)
        libICE.so.6 => /lib/x86_64-linux-gnu/libICE.so.6 (0x00007f4a3ad2f000)
        libGLdispatch.so.0 => /lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007f4a3ac77000)
        libsqlite3.so.0 => /lib/x86_64-linux-gnu/libsqlite3.so.0 (0x00007f4a3ab4e000)
        libwebp.so.6 => /lib/x86_64-linux-gnu/libwebp.so.6 (0x00007f4a3aae4000)
        libzstd.so.1 => /lib/x86_64-linux-gnu/libzstd.so.1 (0x00007f4a3aa3b000)
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f4a3aa10000)
        libjbig.so.0 => /lib/x86_64-linux-gnu/libjbig.so.0 (0x00007f4a3a9ff000)
        libjpeg.so.8 => /lib/x86_64-linux-gnu/libjpeg.so.8 (0x00007f4a3a97a000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f4a3a95e000)
        libXau.so.6 => /lib/x86_64-linux-gnu/libXau.so.6 (0x00007f4a3a958000)
        libXdmcp.so.6 => /lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f4a3a950000)
        libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007f4a3a945000)
        libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007f4a3a92b000)```

**Reply by emgucv** [reply, marked as answer]
Instructions for using the nuget packages on Linux has been updated:
https://www.emgu.com/wiki/index.php/Download_And_Installation#Nuget_2

For latest Raspberry Pi OS (debian bookworm) release, adding the nuget pacakge and installing the dependency 
https://github.com/emgucv/emgucv/blob/4.9.0/platforms/debian/bookworm/apt_install_dependency
is sufficient for using Emgu CV.
